app/controllers/users_controller.rb

```diff
@@ -15,4 +15,12 @@ def destroy
     flash[:success] = 'User deleted successfully.'
     redirect_to users_path
   end
+
+  def make_admin
+    @user = User.find(params[:user_id])
+    @user.role = "admin"
+    @user.save
+    flash[:success] = "#{@user.username} is now an admin."
+    redirect_to users_path
+  end
```

Comment:
This method goes against Rails convention. You should update a user with the role admin in the params.